src/ucp/core/ucp_context.h

```diff
@@ -497,24 +497,23 @@ ucp_memory_detect_internal(ucp_context_h context, const void *address,
     }
 
     status = ucs_memtype_cache_lookup(address, length, mem_info);
-    if (status != UCS_ERR_NO_ELEM) {
-        if (ucs_likely(status != UCS_OK)) {
-            ucs_assert(status == UCS_ERR_NO_ELEM);
-            goto out_host_mem;
-        }
-
-        if ((mem_info->type != UCS_MEMORY_TYPE_UNKNOWN) &&
-            ((mem_info->sys_dev != UCS_SYS_DEVICE_ID_UNKNOWN))) {
-            return;
-        }
-
-        /* Fall thru to slow-path memory type and system device detection by UCT
-         * memory domains. In any case, the memory type cache is not expected to
-         * return HOST memory type.
-         */
-        ucs_assert(mem_info->type != UCS_MEMORY_TYPE_HOST);
+    if (ucs_likely(status == UCS_ERR_NO_ELEM)) {
+        goto out_host_mem;
     }
 
+    ucs_assert(status == UCS_OK);
+
+    if ((mem_info->type != UCS_MEMORY_TYPE_UNKNOWN) &&
+        (mem_info->sys_dev != UCS_SYS_DEVICE_ID_UNKNOWN)) {
+        return;
```

Comment:
maybe move line 517 inside this if and inverse it's condition

Reply:
done

Reply:
done